File: Data Structures & Algorithms/subsets/submission-3.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> result = new ArrayList();
        List<Integer> subset = new ArrayList();
        backtrack(result, subset, nums, 0);
        return result;
    }

    public void backtrack(List<List<Integer>> result, List<Integer> subset, int[] nums, int pos) {
        result.add(new ArrayList<>(subset));
        for (int i = pos; i < nums.length; i++) {
            subset.add(nums[i]);
            backtrack(result, subset, nums, i + 1);
            subset.remove(subset.size() - 1);
        }
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        result = 0
        for i in s:
            if i - 1 not in s:
                curr_streak = 1
                while (i + 1 in s):
                    curr_streak += 1
                    i += 1
                result = max(result, curr_streak)
        return result